<?xml version="1.0"?>
<robot name="HarpoonGun">

  <link name="base_link">
    <!-- Base link using mesh file -->
    <visual>
      <geometry>
        <mesh filename="package://41068_ignition_bringup/urdf/meshes/harpoon_arm/base.STL"/>
      </geometry>
    </visual>
  </link>

  <link name="link1">
    <!-- Barrel -->
    <visual>
      <geometry>
        <mesh filename="package://41068_ignition_bringup/urdf/meshes/harpoon_arm/barrel.STL"/>
      </geometry>
    </visual>
  </link>

  <!-- Joints
  <joint name="joint1" type="revolute">
    <parent link="base_link"/>
    <child link="link1"/>
    <origin xyz="0 0 0.22" rpy="0 0 0"/>
    <axis xyz="1 0 0"/>
    <limit lower="0.0" upper="0.785" effort="10" velocity="1"/>
  </joint> -->

  <!-- Joint connecting base_link to barrel -->
  <joint name="joint1" type="fixed">
    <parent link="base_link"/>
    <child link="link1"/>
    <origin xyz="0 0 0.22" rpy="0 0 0"/> <!-- adjust rpy if needed to make barrel flat -->
  </joint>

</robot>
